packages/nimble-components/src/table/index.ts

```diff
@@ -162,6 +217,61 @@ export class Table<
 
         this.updateTableOptions({ columns: generatedColumns });
     }
+
+    private updateVirtualizer(): void {
+        const options = this.createVirtualizerOptions();
+        if (this.virtualizer) {
+            this.virtualizer.setOptions(options);
+        } else {
+            this.virtualizer = new Virtualizer(options);
+        }
+        this.virtualizer._willUpdate();
+        this.handleVirtualizerChange();
+    }
+
+    private createVirtualizerOptions(): VirtualizerOptions<
+    HTMLElement,
+    HTMLElement
+    > {
+        const rowHeight = parseFloat(controlHeight.getValueFor(this));
+        return {
+            count: this.tableData.length,
+            getScrollElement: () => {
+                return this.viewport;
+            },
+            estimateSize: (_: number) => rowHeight,
+            enableSmoothScroll: true,
+            overscan: 3,
+            scrollToFn: elementScroll,
+            observeElementOffset,
+            observeElementRect,
+            onChange: () => this.handleVirtualizerChange()
+        } as VirtualizerOptions<HTMLElement, HTMLElement>;
+    }
+
+    private handleVirtualizerChange(): void {
+        const virtualizer = this.virtualizer!;
+        this.visibleItems = virtualizer.getVirtualItems();
+        this.allRowsHeight = virtualizer.getTotalSize();
+        // We're using a separate div ('table-scroll') to represent the full height of all rows, and
+        // the row container's height is only big enough to hold the virtualized rows. So we don't
+        // use the TanStackVirtual-provided 'start' offset (which is in terms of the full height)
+        // to translate every individual row, we just translate the row container.
+        let rowContainerTransform = '';
+        if (this.visibleItems.length > 0) {
+            const firstItem = this.visibleItems[0]!;
+            const lastItem = this.visibleItems[this.visibleItems.length - 1]!;
+            if (lastItem.end < this.allRowsHeight) {
+                const offsetY = firstItem.start - this.viewport.scrollTop;
+                rowContainerTransform = `translateY(${offsetY}px)`;
+            }
+        }
+        this.rowContainer.style.transform = rowContainerTransform;
```

Comment:
Why don't we just create an bindable property (like `rowContainerYOffset`), which defaults to 0, like you did for `headerContainerMarginRight`, and then just set the style in the template?

Reply:
Updated